cpu/gd32v/nano/nanostubs.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2017 Ken Rabold
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_fe310
+ * @{
+ *
+ * @file        nanostubs.c
+ * @brief       Implementation of the Newlib-nano for SiFive FE310
+ *
+ * @author      Ken Rabold
+ * @}
+ */
+
+#include <reent.h>
+#include <errno.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/unistd.h>
+
+#include "sched.h"
+#include "thread.h"
+#include "irq.h"
+#include "cpu.h"
+#include "stdio_uart.h"
+
+extern char _sheap;            /* Heap markers from fe310.ld file */
+extern char _eheap;
+char *heap_top = &_sheap + 4;
+
+/**
+ * @brief Initialize the Newlib-nano functions (also forces inclusion of stubs for linking)
+ */
+void nanostubs_init(void)
+{
+    /* STDIO redirected to UART, no line buffering */
+    setvbuf(stdout, NULL, _IONBF, 0);
+    stdout->_write = _write_r;
+}
+
+void _init(void)
+{
+}
+
+void _fini(void)
+{
+}
+
+void *_sbrk(ptrdiff_t incr)
+{
+    unsigned int state = irq_disable();
+    void *res = heap_top;
+
+    /* Allocate memory from heap */
+    if ((heap_top + incr > &_eheap) || (heap_top + incr < &_sheap)) {
+        errno = ENOMEM;
+        res = (void *) -1;
+    }
+    else {
+        heap_top += incr;
+    }
+
+    irq_restore(state);
+    return res;
+}
+
+int _open(const char *name, int flags, int mode)
+{
+    (void) name;
+    (void) flags;
+    (void) mode;
```

Comment:
```suggestion
    (void)name;
    (void)flags;
    (void)mode;
```

Reply:
Removed the whole file, not sure why it was included, probably leftover from some initial iteration :|